### ISSUE TITLE
Value cannot be null. Parameter name: clientSecret

### ISSUE THREAD

**Issue by faridskyman** [OfficeDev/O365-WebApp-MultiTenant#12]
Using VS2015 Community, after following the 4 steps to configuration, i built and run the app and manage to login

![image](https://cloud.githubusercontent.com/assets/2541992/11778029/e6a4621a-a28c-11e5-8dd9-c208328996b6.png)

But when i click on 'My Contact' i got this err:-

![image](https://cloud.githubusercontent.com/assets/2541992/11778059/0878546e-a28d-11e5-8f92-506bc2ac5234.png)

Any idea what i could do? i keen to create an app that can work across multi tenant.
thx

**Comment by faridskyman** [comment]
this post has solved my issue::
https://github.com/OfficeDev/O365-WebApp-MultiTenant/issues/7

But now when i hit 'my contact' i get this

![image](https://cloud.githubusercontent.com/assets/2541992/11778302/73feea08-a28e-11e5-8a91-9789b5f0cfa5.png)


Hitting any of the 2 links gets me

![image](https://cloud.githubusercontent.com/assets/2541992/11778431/2662e140-a28f-11e5-96db-24708500a8a5.png)

**Comment by luisduran365** [comment]
Any luck with this? I have the same issue trying to set this solution up.

**Comment by 811** [comment]
I am having the same problem. When I click on "My Contact" I get this thing:
Failed to read your contacts. Click here to sign in or here to consent app. But both these links give an error.
So can I get a solution for this issue..?

**Comment by dongheechoi** [comment]
Same problem I have. Is this project working only on VS2013? There should be a tutorial about converting to VS2015.

**Comment by faridskyman** [comment]
i went to office 365 admin > administer AD > my app > review permission for the app > and gave anything thing with contacts mentioned permission > tried again and i no longer have the 'failed to read contacts' but when i add contact, gets blank page.. i think that part is not yet developed... *sigh*

**Comment by elegault** [comment]
Has anyone found a solution to this?  I tried creating an MVC app from scratch and added everything necessary to have it work with calendars instead - still the same problem.  I registered it as a different app as well of course.  @faridskyman how were you able to reconfigure the permissions in Azure?  I can't find an option to do that, it just has option to "Get Info", "Remove" and "Report a problem".  Other pages for managing app settings also only have the options to remove, not reconfigure:

https://portal.office.com/account/#apps
https://portal.office.com/myapps

However I am able to change the permissions by going through the "Add Connected Service" wizard again.  Nothing makes a difference though - same "resource cannot be found" errors.  The problem seems to be in ContactsController.Index(), on this line: 

`var dcr = await discClient.DiscoverCapabilityAsync("Contacts");`

It throws an AdalException with the error "Failed to acquire token silently. Call method AcquireToken".

Any ideas anyone?
